# Graded assignment 1

Frank Verbruggen


# Packages

We need the package `tidyverse`

In [1]:
library(tidyverse) 

-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
<U+221A> ggplot2 2.2.1     <U+221A> purrr   0.2.4
<U+221A> tibble  1.4.2     <U+221A> dplyr   0.7.4
<U+221A> tidyr   0.8.0     <U+221A> stringr 1.3.0
<U+221A> readr   1.1.1     <U+221A> forcats 0.3.0
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [2]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2        V3        V4        V5        V6        V7       
1 0.5664549 0.6087127 0.5759644 0.7798237 0.1125907 0.3532992 0.6685180
2 0.2231356 0.7565179 0.2969965 0.6571487 0.4372675 0.9887309 0.1078339
  V8          V9        V10       V11       V12        V13        V14      
1 0.008242477 0.9777053 0.1519341 0.1290944 0.09088921 0.01931911 0.2498563
2 0.119680366 0.4424443 0.5389883 0.5990740 0.39655346 0.46569127 0.7710350
  V15       V16       V17        V18       V19       V20      
1 0.4691357 0.1053102 0.29881490 0.9267388 0.8094012 0.3835071
2 0.9060054 0.7714446 0.04974283 0.4276346 0.1886059 0.6123911

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [5]:
df2 <- df1 %>% 
    gather(Key, Value, V1:V20)
head(df2)

Key Value    
1 V1  0.5664549
2 V1  0.2231356
3 V2  0.6087127
4 V2  0.7565179
5 V3  0.5759644
6 V3  0.2969965

## Assignment 2

We will generate some data:

In [6]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [9]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [10]:
df5 <- full_join(df3, df4, by = 'id')
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook